src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1218,8 +1309,14 @@ namespace cryptonote
       return false;
     }
     add_new_block(b, bvc);
-    if(update_miner_blocktemplate && bvc.m_added_to_main_chain)
-       update_miner_block_template();
+
+    if (bvc.m_added_to_main_chain)
+    {
+      m_deregister_vote_pool.remove_expired_votes(get_current_blockchain_height());
+
+      if(update_miner_blocktemplate)
+         update_miner_block_template();
+    }
```

Comment:
All good

Reply:
I'm going to move this into blockchain::handle_block_to_main_chain() because at the end of the function there's a call to m_tx_pool.on_blockchain_inc(..) which we sort of agreed that this was likely where the txpool is meant to flush out tx's that are now invalid because of the new block (but isn't at the moment). So similarly we can flush out deregister votes in the pool in the same spot with the added bonus we also have access to the TX's so I don't need to re-parse the block data again.

It means adding to the blockchain constructor a reference to the deregister vote pool though and whatever else it takes to get that compiling (blackball etc).

Reply:
Fair enough, let's just get this finished